processing/src/main/java/org/apache/druid/frame/file/FrameFileHttpResponseHandler.java

```diff
@@ -105,20 +107,22 @@ private ClientResponse<FrameFilePartialFetch> response(
       return ClientResponse.finished(clientResponseObj);
     }
 
-    final byte[] chunk = new byte[content.readableBytes()];
-    content.getBytes(content.readerIndex(), chunk);
+    if (content != null) {
+      final byte[] chunk = new byte[content.readableBytes()];
+      content.getBytes(content.readerIndex(), chunk);
```

Comment:
it might be possible to optimize this later one by incrementing the reference count to the underlying ByteBuf and passing that around directly